src/endian/stream.hpp

```diff
@@ -15,16 +15,13 @@ class stream
 {
 public:
 
-    /// Creates an endian stream on top of a pre-allocated buffer of the
-    /// specified size.
+    /// Creates an endian stream used to track a buffer of the specified size.
     ///
-    /// @param buffer a pointer to the buffer
     /// @param size the size of the buffer in bytes
     stream(uint32_t size) :
-        m_size(size),
-        m_position(0)
+        m_size(size)
     {
-        assert(m_size > 0);
+        assert(size > 0 && "Tracking buffer of size zero is not valid");
```

Comment:
I think that is a bit too black and white. The is no reason IMO why you can check two things at once. The user in this case is a developer - which will totally be able to see that `size == m_size`, when he goes to the offending line where the assert is placed.